### ISSUE TITLE
Notification sent from satellites when the parent zone is unavailable

### ISSUE THREAD

**Issue by winem** [Icinga/icinga2#5298]
Hi, 

I'm currently refactoring the whole monitoring environment and had an idea for an enhancement regarding notifications. 

## Current situation:
In the current case, there is master cluster in the companies network (internal server room) with 3 childs (satellites in the same zone) out there. Those 3 childs are hosted at different providers. Now, there is only one bottleneck and SPOF: The uplink (even though it's redundant) of the companies server room. 

## Impact: 
The most critical impact is that we would not receive any information / notification in case of an outtage at any customer environment out there. 

## Possible solution: 
Extend the notifications module and add a parameter like "if_parent_down_only". There might be 2 options to achieve this: 
1. the notifications feature at the satellites will be enabled the whole time and icinga2 decides case-by-case whether it should send the notification or not (depending on the availability of the parent zone)
2. the satellite enables the notification feature and reloads the configuration as soon as it detects that it's parent is down. 

When a notification has been sent from a satellite, the master should not send a new one as soon as it receives the replay log from the satellite. 

In my opinion, this would be a very valuable feature. I do not have that much expertise with C++, so that I can not provide an ad-hoc solution but feel free to involve me during the tests.

**Comment by marcofl** [comment]
maybe this goes in the same direction: https://github.com/Icinga/icinga2/issues/5381
for your case, I think this could help:
by setting `zone` of the satellites host object to the master zone and using `cluster_zone`host check command, you'll see when the satellite is offline, if in your case, the satellite is remote. if only the clients are remote, then you can do the same on the satellites to notice if the clients are down.

**Comment by winem** [comment]
Hi, 

I will test your first suggestion within the next two weeks since I already have some projects with tough timelines and I'll be on vacation for one week. At the moment, I'm not sure if this works when the satellites can not connect to the master. In our case, the master (satellites do not have the host set for master in the zone file) connects to the satellites. Anyway, I'll give it a try as soon as I have time for it. 

I don't get how an if statement could help here because it's evaluated during the restart / config reload of icinga2 and won't change "on the fly", if I'm right. So, a config example is appreciated.

**Comment by Al2Klimov** [comment]
@N-o-X Shall we wait for feedback on "test your first suggestion" or close this for the same reason as https://github.com/Icinga/icinga2/issues/5381#issuecomment-311595508 ?

**Comment by winem** [comment]
@Al2Klimov, I'm sorry for the lack of response back then. From my side, this issue can be closed as I am no longer working with Icinga.Unless you see a general need for this feature.

We got away with the cluster zone check and notifications just for this specific service and an improved infrastructure for the master instances so that we reduced the risk for a downtime on the master nodes to an absolute minimum and it happened actually just once for < 20 minutes in the past 2 years.
